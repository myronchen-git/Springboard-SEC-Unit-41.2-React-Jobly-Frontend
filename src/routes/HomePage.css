.HomePage__welcome-back-message {
  font-size: x-large;
}
