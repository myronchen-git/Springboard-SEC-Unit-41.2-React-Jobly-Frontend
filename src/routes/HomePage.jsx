import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Button } from 'reactstrap';

import './HomePage.css';

// ==================================================

/**
 * Displays the home page.  Displays different views depending on whether a
 * user is logged in or not.
 *
 * @param {Object} props - React component properties.
 * @param {String} props.username - Username of the logged in user, otherwise
 *   falsy.
 */
function HomePage({ username }) {
  return (
    <main className="HomePage">
      <h1>Jobly</h1>
      <p>All the jobs in one, convenient place.</p>
      {username ? (
        <div className="HomePage__welcome-back-message">
          <p>Welcome Back, {username}!</p>
        </div>
      ) : (
        <div>
          <Link className="m-2" to="/login">
            <Button color="light">Log In</Button>
          </Link>
          <Link className="m-2" to="/signup">
            <Button color="light">Sign Up</Button>
          </Link>
        </div>
      )}
    </main>
  );
}

// ==================================================

HomePage.propTypes = {
  username: PropTypes.string,
};

export default HomePage;
