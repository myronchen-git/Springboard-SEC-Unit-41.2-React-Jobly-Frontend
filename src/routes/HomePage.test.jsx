import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';

import HomePage from './HomePage.jsx';

// ==================================================

describe('CompanyPage', () => {
  const username = 'testuser';

  it('renders.', () => {
    render(
      <MemoryRouter>
        <HomePage username={username} />
      </MemoryRouter>
    );
  });

  it('matches snapshot.', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HomePage username={username} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays correct text when logged in.', () => {
    // Act
    const { queryByText, getByText } = render(
      <MemoryRouter>
        <HomePage username={username} />
      </MemoryRouter>
    );

    // Assert
    expect(getByText('Welcome Back', { exact: false })).toBeVisible();
    expect(queryByText('Log In')).not.toBeInTheDocument();
    expect(queryByText('Sign Up')).not.toBeInTheDocument();
  });

  it('displays correct text when not logged in.', () => {
    const username = undefined;

    // Act
    const { queryByText, getByText } = render(
      <MemoryRouter>
        <HomePage username={username} />
      </MemoryRouter>
    );

    // Assert
    expect(
      queryByText('Welcome Back', { exact: false })
    ).not.toBeInTheDocument();
    expect(getByText('Log In')).toBeVisible();
    expect(getByText('Sign Up')).toBeVisible();
  });
});
