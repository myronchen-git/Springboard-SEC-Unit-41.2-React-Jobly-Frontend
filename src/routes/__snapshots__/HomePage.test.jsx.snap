// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyPage > matches snapshot. 1`] = `
<DocumentFragment>
  <main
    class="HomePage"
  >
    <h1>
      Jobly
    </h1>
    <p>
      All the jobs in one, convenient place.
    </p>
    <div
      class="HomePage__welcome-back-message"
    >
      <p>
        Welcome Back, testuser!
      </p>
    </div>
  </main>
</DocumentFragment>
`;
